### DISCUSSION TITLE
How to convert variable of CONSTANT, MONOMIAL to FIRST, LAGRANGE in auxkernel?

### DISCUSSION THREAD

**Discussion by SomnusYu** [idaholab/moose#27130] [Q&A General] [answered]
Hi all,

I notice that the type of flow channel variables in THM are `CONSTANT`, and `MONOMIAL`, which is in a stepped shape. I would like to show them as continuous functions, like the variable type of `FIRST`, and `LAGRANGE`. Is there any auxkernel to do this? Or I must use a postprocessor?
Thanks!

**Comment by GiudGiud** [comment, marked as answer]
Hello

You can use the ProjectionAux to make L2-norm minimizing (in most cases) projections between variable types. It can go from CONSTANT MONOMIAL to FIRST LAGRANGE with an auxiliary variable for the FIRST LAGRANGE 

Guillaume

**Reply by SomnusYu** [reply]
Thanks, I will have a try!
